processing/src/main/java/io/druid/segment/incremental/SpatialDimensionRowTransformer.java

```diff
@@ -46,19 +46,27 @@
  */
 public class SpatialDimensionRowTransformer implements Function<InputRow, InputRow>
 {
-  private static final Joiner JOINER = Joiner.on(",");
-  private static final Splitter SPLITTER = Splitter.on(",");
+  private final Map<String, Joiner> joinerCache;
```

Comment:
Maybe encapsulate joiner and splitter in `SpatialDimensionSchema` and add methods like `SpatialDimensionSchema.isJoinedValueValid()` and another one, whose implementation is `joiner.join(spatialDimVals)`? It would allow to avoid making those maps, populating and querying them.